packages/lesswrong/server/callbacks/sunshineCallbackUtils.tsx

```diff
@@ -126,23 +126,65 @@ function handleAutomodAction(triggerAction: boolean, userId: string, actionType:
   }
 }
 
+/**
+ * WARNING: assumes that the input actions are already sorted by createdAt descending
+ */
+function getLastActionEndedAt(actions: DbModeratorAction[], actionType: DbModeratorAction['type']) {
+  return actions.find(action => action.type === actionType)?.endedAt;
+}
+
+function getUnmoderatedContent(content: (DbPost | DbComment)[], lastActionEndedAt?: Date | null) {
+  return lastActionEndedAt
+    ? content.filter(item => item.postedAt > lastActionEndedAt)
+    : content;
+}
+
+/**
+ * recently downvoted content:
+ * - if active, disable if no longer meets condition
+ * - if inactive, condition should only check content (both posts & comments) from after the last endedAt
+ * 
+ * low average karma:
+ * - if active, disable if no longer meets condition
+ * - if inactive, condition should only check content (whichever one matches the action type) from after the last endedAt
+ */
 export async function triggerAutomodIfNeededForUser(user: DbUser) {
   const userId = user._id;
 
+
   const lowUserKarma = user.karma < -5;
   handleAutomodAction(lowUserKarma, userId, NEGATIVE_KARMA_USER_ALERT);
 
-  const [latestComments, latestPosts] = await Promise.all([
+  const [userModeratorActions, latestComments, latestPosts] = await Promise.all([
+    // Sort by createdAt descending so that `.find` returns the most recent one matching the condition
+    ModeratorActions.find({ userId }, { sort: { createdAt: -1 } }).fetch(),
     Comments.find({ userId }, { sort: { postedAt: -1 }, limit: 20 }).fetch(),
     Posts.find({ userId, isEvent: false, draft: false }, { sort: { postedAt: -1 }, limit: 20 }).fetch()
   ]);
 
+  // Get the `endedAt` of the most recently-created action of each type
+  // We don't care about the distinction between an active action with no `endedAt` and the user not having that type of action at all
+  // We'll check that distinction later in `triggerModerationAction`, if necessary
+  // Also, we want to be able to disable active actions on the basis of recent content getting upvoted
+  // That would be impossible if we filtered it out
+  const downvotedContentActionEndedAt = getLastActionEndedAt(userModeratorActions, RECENTLY_DOWNVOTED_CONTENT_ALERT);
+  const lowAvgKarmaCommentEndedAt = getLastActionEndedAt(userModeratorActions, LOW_AVERAGE_KARMA_COMMENT_ALERT);
+  const lowAvgKarmaPostEndedAt = getLastActionEndedAt(userModeratorActions, LOW_AVERAGE_KARMA_POST_ALERT);
+
   const voteableContent = [...latestComments, ...latestPosts].sort((a, b) => b.postedAt.valueOf() - a.postedAt.valueOf());
+
+  // Remove the most recent content item for each rule
+  // Since posts & comments start by default without much karma, they artificially down-weight averages
+  latestComments.pop();
+  latestPosts.pop();
```

Comment:
@s-cheng Whoops, yes, good catch.

Responding to the above question on motivation: the pattern that we've observed when using the new moderation dashboard is that a lot of users who get flagged with `LOW_AVERAGE_KARMA_COMMENT_ALERT` no longer have that moderator action active on them by the time we're reviewing them, since they'll have gotten some more upvotes.   If we did see a user who made e.g. 10 comments on a bunch of different posts, and none of them got upvoted, that is actually a warning flag that we might want keep an eye on that user to see if they're commenting at a high rate but with mediocre comments.

But if we look at a user like that, and it seems like their comments are fine, just aren't getting much voting engagement for other reasons, we don't really want the next comment (or two) that they make to immediately flag them for review again, as would happen by default if they just barely pulled out of the low average karma threshold with one of their recent comments, and another unvoted-on comment would push them back under it.

Coming up with rules that avoid giving us too many false positives without missing a bunch of cases is proving to be a bit tricky and I can imagine that the EA forum might want different heuristics (as already exists for comment-specific moderator actions, due to a suggestion from a couple weeks ago by @jpaddison3).

Reply:
Cool makes sense! I had thought that you all would care more about missing users who should have been caught, but I didn't realize that you were already getting a lot of false positives. Though it feels to me that the `RECENTLY_DOWNVOTED_CONTENT_ALERT` should be separate, since you probably get less false positives with that one, and care more about not missing those users?

Reply:
It's true that it's a stronger signal, but if we get a user that has 2 downvoted comments in the last 10 comments, and then we review them and decide we don't need to take action, we'll probably have looked at all 10 comments rather than just the 2 downvoted ones.  So when `RECENTLY_DOWNVOTED_CONTENT_ALERT` gets ended by the automod (either because the downvoted comments got upvoted back to net-positive, or because those comments fell out of the last-10-comments window for that user), we either don't want to consider those comments still in the window, or we won't have any remaining overlap anyways.